app/js/components/service_form.ts

```diff
@@ -0,0 +1,104 @@
+    $privacyQuestionsStatusContainer: JQuery;
+    $contractSignedToggle: JQuery;
+    $contractSignedContainer: JQuery;
+
+    /**
+     * @param $privacyQuestionToggle The selectbox to toggle the privacy question on
+     * @param $privacyQuestionContainer The form-row twith the privacy enabled question to show/hide
+     * @param $contractSignedToggle The container with radio buttons to toggle on
+     * @param $contractSignedContainer The form-row with the signed-contract questions to show/hide
+     */
+    constructor(
+        $privacyQuestionToggle: JQuery,
+        $privacyQuestionContainer: JQuery,
+        $contractSignedToggle: JQuery,
+        $contractSignedContainer: JQuery
+    ) {
+        this.$privacyQuestionsEnabledToggle = $privacyQuestionToggle;
+        this.$privacyQuestionsStatusContainer = $privacyQuestionContainer;
+        this.$contractSignedToggle = $contractSignedToggle;
+        this.$contractSignedContainer = $contractSignedContainer;
+    }
+
+    /**
+     * Init the eventhandlers on the elements
+     */
+    registerEventHandlers() {
+        // Init the privacy questions answered toggle
+        this.initTogglePrivacyQuestionsAnsweredStatusField();
+
+        // Init the contract signed toggle
+        this.initToggleContractSignedStatusField();
+    }
+
+    /**
+     * Disable the contract signed field when the type of service is for an institute
+     */
+    private initToggleContractSignedStatusField()
+    {
+        let toggleContractSigned = () => {
+            let serviceTypeValue = this.$contractSignedToggle.find(':checked').val();
+            if (serviceTypeValue === "institute") {
+                this.showElement(this.$contractSignedContainer.parent());
+            } else {
+                this.hideElement(this.$contractSignedContainer.parent());
+            }
+        };
+
+        this.$contractSignedToggle.on('change', toggleContractSigned);
+        toggleContractSigned();
+    }
+
+    /**
+     * Disable the privacy questions answered field when the privact question enabled checkbox is unchecked
+     */
+    private initTogglePrivacyQuestionsAnsweredStatusField()
+    {
+        let togglePrivacyQuestions = () => {
+            if (this.$privacyQuestionsEnabledToggle.is(':checked')) {
+                this.showElement(this.$privacyQuestionsStatusContainer.parent());
+            } else {
+                this.hideElement(this.$privacyQuestionsStatusContainer.parent());
+            }
+        };
+        this.$privacyQuestionsEnabledToggle.on('change', togglePrivacyQuestions);
+        togglePrivacyQuestions()
+    }
+
+    /**
+     * Helper method to hide an element
+     */
+    private hideElement($element: JQuery) {
+        $element.addClass('hidden');
+        $element.hide(200);
+
+    }
+
+    /**
+     * Helper method to show an element
+     */
+    private showElement($element: JQuery) {
+        $element.removeClass('hidden');
+        $element.show(200);
```

Comment:
I thought jQuery animations where dropped in favor of CSS transitions?

Reply:
I've looked for a pure css solution but you would require a `hidden` and a `show` state and for each state a `enter` and a `leave`. Because if you add a `display: none` there is no animation. I however didn't want to be dependent of the animations for testing so I've added a `hidden` class to the element to test the state for testing. I've also removed the css for the `hidden` class, because that isn't used anymore.